src/tls13cert.rs

```diff
@@ -45,33 +43,57 @@ pub(crate) fn asn1_error<T>(err: Asn1Error) -> Result<T, Asn1Error> {
 // Long form length
 // * Must be used when the length is 128 or greater
 // * XXX: We do not accept lengths greater than 32-bit.
+#[hax_lib::ensures(|result| match result {
+    Result::Ok(_) => b.len() >= offset && b.len() - offset >= len,
+    _ => true })]
 fn long_length(b: &Bytes, offset: usize, len: usize) -> Result<usize, Asn1Error> {
     if len > 4 {
         asn1_error(ASN1_SEQUENCE_TOO_LONG)
     } else {
-        let mut u32word = [U8(0); 4];
-        u32word[0..len].copy_from_slice(&b[offset..offset + len]);
-        Ok(u32_from_be_bytes(u32word).declassify() as usize >> ((4 - len) * 8))
+        if b.len() >= offset + len {
```

Comment:
I'm not sure if this is how it's used. This requires the caller to pass in only the exact slice that's read. I don't think that's how this should be defined.